File: truss-compiler/src/main/java/truss/compiler/preprocessor/PreProcessorStringStream.java
package truss.compiler.preprocessor;

import org.antlr.runtime.ANTLRStringStream;

import java.util.Set;

public class PreProcessorStringStream extends ANTLRStringStream {
    public PreProcessorStringStream(String input, Set<String> defines) throws PreProcessException {
        super(input);

        init(defines);
    }

    public PreProcessorStringStream(char[] data, int numberOfActualCharsInArray, Set<String> defines) throws PreProcessException {
        super(data, numberOfActualCharsInArray);

        init(defines);
    }

    private void init(Set<String> defines) throws PreProcessException {
        data = PreProcessor.process(data, n, defines);
        n = data.length;
    }
}

File: Truss.Compiler/PreProcessor/PreProcessorStringStream.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Antlr.Runtime;

namespace Truss.Compiler.PreProcessor {
    public class PreProcessorStringStream : ANTLRStringStream {
        public PreProcessorStringStream(ErrorList errors, string input, IEnumerable<string> defines)
            : base(input) {
            Init(errors, defines);
        }

        public PreProcessorStringStream(ErrorList errors, char[] data, int numberOfActualCharsInArray, IEnumerable<string> defines)
            : base(data, numberOfActualCharsInArray) {
            Init(errors, defines);
        }

        private void Init(ErrorList errors, IEnumerable<string> defines) {
            data = PreProcessor.Process(errors, data, n, defines);
            n = data.Length;
        }
    }
}
